rollup/circuitcapacitychecker/libzkp/src/lib.rs

```diff
@@ -0,0 +1,81 @@
+#![feature(once_cell)]
+
+pub mod checker {
+    use crate::utils::{bool_to_int, c_char_to_vec};
+    use libc::c_char;
+    use std::cell::OnceCell;
+    use std::panic;
+    use types::eth::BlockTrace;
+    use zkevm::capacity_checker::CircuitCapacityChecker;
+
+    static mut CHECKER: OnceCell<CircuitCapacityChecker> = OnceCell::new();
+
+    /// # Safety
+    #[no_mangle]
+    pub unsafe extern "C" fn new_circuit_capacity_checker() {
+        env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("debug"))
+            .format_timestamp_millis()
+            .init();
+
+        let c = CircuitCapacityChecker::new();
+        CHECKER.set(c).unwrap();
+    }
+
+    /// # Safety
+    #[no_mangle]
+    pub unsafe extern "C" fn reset_circuit_capacity_checker() {
+        CHECKER.get_mut().unwrap().reset()
+    }
+
+    /// # Safety
+    #[no_mangle]
+    pub unsafe extern "C" fn apply_tx(tx_traces: *const c_char) -> c_char {
+        let tx_traces_vec = c_char_to_vec(tx_traces);
+        let traces = serde_json::from_slice::<BlockTrace>(&tx_traces_vec).unwrap();
```

Comment:
when doing benchmark, we may need a result, where AFTER traces deserialized, just return 0 in this function to skip `estimate_circuit_capacity`.

With this result, we can better analyze which part is bottleneck.   For example, if we get 3tps for erc20 transfer when capacity checker enabled, then we re-bench with estimate_circuit_capacity function skipped (return 0 simply), then 
(1) if we get 5 tps, we can conclude the trace generation performance is quite bad, we need to improve that.
(2) if we get 30 tps, we can conclude the  trace generation performance is good, we need to improve rust codes.

Reply:
ok

Reply:
> traces deserialized

@lispc I tested the API on my laptop not on dev-machine, but it seems there are performance issue:

---

blocktimestamp with circuit API:
initBalance 0x6492908a -> 1687326858
createPair  0x6492908e -> 1687326862
addLiq        0x64929092 -> 1687326866
swap           0x64929097 -> 1687326871
swap           0x649290bc -> 1687326908

---

blocktimestamp skipping circuit API call (but with traces deserialization):
initBalance 0x6492a494 -> 1687331988
createPair  0x6492a497 -> 1687331991
addLiq        0x6492a49a -> 1687331994
swap           0x6492a49d -> 1687331997
swap           0x6492a4a3 -> 1687332003

---

blocktimestamp without circuit API at all:

initBalance 0x6492ade4 -> 1687334372
createPair  0x6492ade7 -> 1687334375
addLiq        0x6492adea -> 1687334378
swap           0x6492aded -> 1687334381
swap           0x6492adf0 -> 1687334384

Reply:
@lispc if you want to test
you can build the docker using `docker build -t scrolltech/l2geth:XXX ./`
then run with https://github.com/scroll-tech/trace-dumper (need to change https://github.com/scroll-tech/trace-dumper/blob/dev/docker/l2geth/Dockerfile#L1 to `scrolltech/l2geth:XXX`)